lib/node-package-manager.ts

```diff
@@ -50,128 +26,51 @@ export class NodePackageManager implements INodePackageManager {
 		});
 		return future;
 	}
-
-	public install(packageName: string, opts?: INpmInstallOptions): IFuture<string> {
-		return (() => {
-			this.$lockfile.lock().wait();
-
-			try {
-				var packageToInstall = packageName;
-				var pathToSave = (opts && opts.pathToSave) || npm.cache;
-				var version = (opts && opts.version) || null;
-
-				return this.installCore(packageToInstall, pathToSave, version).wait();
-			} catch(error) {
-				this.$logger.debug(error);
-				this.$errors.fail("%s. Error: %s", NodePackageManager.NPM_LOAD_FAILED, error);
-			} finally {
-				this.$lockfile.unlock().wait();
-			}
-
-		}).future<string>()();
-	}
-
-	public getLatestVersion(packageName: string): IFuture<string> {
-		var future = new Future<string>();
-
-		npm.commands["view"]([packageName, "dist-tags"], [false], (err: any, data: any) => { // [false] - silent
-			if(err) {
-				future.throw(err);
-			} else {
-				var latestVersion = _.first(_.keys(data));
-				this.$logger.trace("Using version %s. ", latestVersion);
-
-				future.return(latestVersion);
-			}
-		});
-
-		return future;
+	
+	public install(packageName: string, pathToSave: string, config?: any): IFuture<any> {
+		return this.loadAndExecute("install", [pathToSave, packageName], { config: config });
 	}
-
-	public getCachedPackagePath(packageName: string, version: string): string {
-		return path.join(npm.cache, packageName, version, "package");
+	
+	public uninstall(packageName: string, config?: any): IFuture<any> {
+		return this.loadAndExecute("uninstall", [[packageName]], { config: config });
```

Comment:
why do you have double [ around the package name, shouldn't it be [packageName]?